File: descripe/imapp_modisl2/src/leocat/src/MOD_PRAlg29/trispc.c
/*
!C ********************************************************************
!Description:

  Float function trispc.c    

  Regression function relating the pw values to the 8-11um btdif.
  The difference is related to the amount of pw in the atmosphere
  due to the weak water vapor lines present in this spectral
  region.  The regressions were determined from actual HIRS data
  and provide thresholds that can be compared to measured 8-11
  differences.

!Input arguments:
  tdf1      measured 8-11 micron BTD

!Output arguments:
  none

  Output value returned through function call.

!Revision History:
  Original from MODIS Cloud Mask FORTRAN code developed at CIMSS,
  UW-Madison (trispc.f).

  Converted to C
  R. Frey           05/2007


!Team-unique Header:

!References and Credits:

!END ******************************************************************/

/* Includes */

#include <stdio.h>


float trispc(float tdf1)


{

    float a1, a2, a3, a4;
    float tri_thr;
    float x;

/*  Regression coefficients */
    a1 =  2.7681;
    a2 = -3.7290;
    a3 =  1.0540;
    a4 = -0.1020;

    x = tdf1;
    tri_thr = a1 + x * (a2 + x * (a3 + x * a4));

    return tri_thr;

}

File: descripe/imapp_modisl2/src/cloudmask/src/trispc.f
      real function trispc(tdf1)

      implicit none
      save

c!F77 ************************************************************
c
c!Description:
c ... regression function relating the pw values to the 8-11um btdif.
c ... The difference is related to the amount of pw in the atmosphere
c ... due to the weak water vapor lines present in this spectral
c ... region.  the regressions were determined from actual hirs
c ... data, and provides a threshold for clear which can be compared
c ... to actual 8-11 differences
c
c!Input Parameters:
c tdf1      11-12 micron BTDIF observed
c
c!Output Parameters: None
c 
c!Revision History:
c
c!Team-unique Header:
c
c    This software is developed by the MODIS Science Data Support Team
c    for the National Aeronautics and Space Administration,
c    Goddard Space Flight Center, under contract NAS5-32373.
c
c!References and Credits:
c See Cloud Mask ATBD-MOD-06.
c
c!Design Notes:
c   Returns the function value trispc as the 8-11 micron calculated
c    clear value.
c
c!END****************************************************************
c
c ...
c ... scalar arguments ..
      real tdf1
c ...
c ... local scalars ..
c ... variables: a's regression coefficients
      real a1,a2,a3,a4,x
c ...
c ... data statements ..
c ... coefficients for hirs data
c     data a1 /-1.7681/, a2 /-3.729/ ,a3 /1.054/, a4 /-0.102/
      data a1 /2.7681/, a2 /-3.729/ ,a3 /1.054/, a4 /-0.102/

      x = tdf1
c     trispc = a1 + a2*tdf1 + a3*(tdf1*tdf1) + a4*(tdf1**3) 
      trispc = a1 + x*(a2 + x*(a3 + x*a4))
c ...
      return
      end
